src/ostrowski/rationals/bounded.lean

```diff
@@ -0,0 +1,364 @@
+
+import analysis.special_functions.exp_log
+import analysis.special_functions.pow
+import data.real.cau_seq
+
+import ostrowski.rationals.basic
+import abvs_equiv
+
+open list is_absolute_value
+
+lemma prime_norm_lt_one_of_bounded_padic (abv : ℚ → ℝ) [habv : is_absolute_value abv]
+      (hnontriv: abv ≠ trivial_abs)
+      (all_nat_le_one: (∀ z : ℕ, abv z ≤ 1)):
+      ∃ p: ℕ, prime p ∧ abv p < 1 :=
+begin
+  by_contra,
```

Comment:
the next step can be merged by just adding !: by_contra!

Reply:
(no joke)

Reply:
It's actually `contrapose!`.

Reply:
(I would have liked to have `by_contra!`, should we open an issue in mathlib ?